scripts/west_commands/runners/nrfutil.py

```diff
@@ -82,6 +82,20 @@ def _insert_op(self, op):
         self._ops.append(op)
 
     def _exec_batch(self):
+        # Make sure nrfutil is recent enough
+        out = self._exec(['--version'])
+        version = tuple(int(x) for x in out[0]['data']['version'].split('.'))
+        if version < (2, 1, 1):
+            raise RuntimeError('nrfutil version 2.1.1 or later required, run `nrfutil upgrade`.')
+
+        # Make sure the device is in the expected family
+        out = self._exec(['device-info', '--serial-number', f'{self.dev_id}'])
+        for entry in out:
+            if entry['type'] == 'info':
+                detected_family = entry['data']['devices'][0]['deviceInfo']['jlink']['deviceFamily']
```

Comment:
maybe I should just search inside the raw string instead? this is going to be super-brittle, right?

Reply:
This should be stable, the output format will only change with a major version bump.